### DISCUSSION TITLE
Running installation under Rosetta still tries to install to /opt/homebrew

### DISCUSSION THREAD

**Discussion by aschneid75** [Homebrew/discussions#502] [Getting started]
Hello All,

I have a new M1 Mac Mini and have the Arm64 version of Homebrew installed just fine.  Thanks for everyone's hard work on this...very impressive how quickly the amount of work that has been.  However, there are a few apps that I would like to run that have not had their formula ported over, so I would like to have a Rosetta based install.

I use iTerm 2 Beta, which is a native Arm build.  I've tried both copying the iTerm app and setting the copy to run under Rosetta and I've also tried the arch -X86_64 pathway to get Homebrew Intel installed under Rosetta.  Every time I try to run the installer, I get a prompt for my password and it attempts to install in /opt/homebrew instead of /usr/local.  When I run the arch command, it says i386.  However, the uname -m says arm64 regardless, so I understand why the shell script sets it to install to /opt/homebrew.

As I'm writing this, I'm wondering if the issue is the native Arm64 build of iTerm.  Or can we just not run under Rosetta any longer?

Thanks,
Andrew

**Comment by aschneid75** [comment]
I just tried in the default Terminal, running:

arch -X86_64 /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

Got the same password prompt and then it tried installing into /opt/homebrew.  I do have:

eval $(/opt/homebrew/bin/brew shellenv)

in my .zshrc file, so wasn't sure if that's causing some kind of override or something.

**Comment by gromgit** [comment]
Looking at the current installer script:
* it always installs to `/opt/homebrew` on ARM-based Macs
* this is not configurable

**Reply by chobotnice007** [reply]
> this is not configurable
Why? You can modify script.

**Reply by SMillerDev** [reply]
And you'll break everything it creates. This is a bad idea.

**Reply by aschneid75** [reply]
@SMillerDev why would it break?  The Intel binaries are configured to go to /usr/local, so I am still not sure why I can't have both a native install and a Rosetta install?  I literally have one or two apps I want to install under Rosetta, and they run via Launchctl, so I can control their paths.

**Reply by SMillerDev** [reply]
Manually modifying the install script paths is going invalidate most assumptions the script makes. There is/should be a way to install Homebrew in Rosetta and without, and if that doesn't work we should figure out that issue. Manually modifying the install script isn't likely going to fix your issue.

**Reply by aschneid75** [reply]
OK, I understand what you are saying.  I thought you were implying that having the dual install isn't possible any longer.  

I primarily want Grafana installed, and Influx when the 2.0 version is supported. I am already manually installing Influx, so I may just do that for Grafana too.  Or see if installing via source works.

**Comment by aschneid75** [comment]
Yes, but it uses uname -m to determine the architecture.  If I run arch
-X86_64 /bin/bash and issue a uname -m it shows i386. So, I am confused as
to why I cannot get it to install the Intel based Homebrew.

**Comment by hein-j** [comment]
Hi, I don't know if this would help, but I have the arm-native installed in opt, and I could get a second version installed into usr/local through the following steps:
- quit default terminal
- restart it with Rosetta (check box in 'Get Info')
- run the command on brew.sh without any additional prefixes.

**Reply by aschneid75** [reply]
Thanks,

Yes, that is how I have tried it multiple times, in addition to also using the arch way as well.  For some reason, on my machine, it always tries to install the Arm64 version.
